protocols/kad/src/protocol.rs

```diff
@@ -422,7 +422,7 @@ fn resp_msg_to_proto(kad_msg: KadResponseMsg) -> proto::Message {
             key: key.to_vec(),
             record: Some(proto::Record {
                 key: key.to_vec(),
-                value,
+                value: value.to_vec(),
```

Comment:
Same as with the gossipsub branch, I am a bit doubtful that this actually yields improvements because we are copying here again.

Reply:
Hmmm, the knock on is that we're reading from the `Record` which is and can be cloned in kad and user space. So we'll end up allocating to just have vecs in here ( if the Record contains `Bytes` for the `value` which I think it should...).

That comes in 
```
HandlerIn::PutRecordRes {
                    key: record.key,
                    value: record.value, // <- we'd have to convert _from_ Bytes here,
                    request_id,
                },```

Reply:
Sorry the above was for another comment response. 


We are still having to convert to `vec` here, but that should become the _only_ clone, vs a variety of `clone`ing that happens in the kad layers at the moment. 

So this should still yield reduce mem.